package.json

```diff
@@ -10,8 +10,13 @@
     "type": "git",
     "url": "https://github.com/ampproject/amphtml.git"
   },
+  "bin": {
+    "amp": "./amp.js",
+    "gulp": "./gulp-deprecated.js"
+  },
   "scripts": {
-    "preinstall": "node build-system/common/check-package-manager.js"
+    "preinstall": "node build-system/common/check-package-manager.js",
+    "postinstall": "npm un -g gulp-cli --ignore-scripts && npm i -g --ignore-scripts"
```

Comment:
`-g`? Are we telling people to not have `gulp` in their global setup anymore? 😅

Reply:
Yes, we do need to remove the global `gulp-cli` in order to seamlessly redirect invocations of `gulp` to the `amp` runner via the `gulp-deprecated.js` wrapper.

Trying to install the wrapper in the presence of a global `gulp-cli` installation shows the error below. (Redirects won't work.)

![image](https://user-images.githubusercontent.com/26553114/111515150-d52efc80-8728-11eb-9cdd-481a9e7ef86a.png)

If there is no global `gulp` / `gulp-cli` binary, we will abruptly break all usage of `gulp` with a message like this:

![image](https://user-images.githubusercontent.com/26553114/111515534-4969a000-8729-11eb-8640-1cd4b3383cc4.png)

I'll see if I can find a way to replace `gulp-deprecated.js` with a wrapper `gulpfile.js` that redirects individual tasks to the `amp` runner. (Not sure if such a thing is even possible.)

Reply:
This appears to remove global gulp installations, without informing the user.

What if the user installed a global gulp installation for other reasons?

Reply:
I poked around, and there is a slightly less invasive way to this via a node script.
- If there is a global install of `gulp-cli`, print a message saying we can't install the redirect from `gulp` to `amp`. In this scenario, the user has two choices:
  - Uninstall `gulp-cli` and use our redirect
  - Stop using `gulp` commands for AMP development
- If there isn't a global `gulp-cli`, install the redirect and `gulp` commands will seamless pass through to `amp`.

Obviously, this isn't as perfectly backwards compatible, but it no longer forces the user not to use a global `gulp-cli`.

**Yet another consideration:** What if the user didn't have a global `gulp-cli`, but wants to install it _after_ we've installed the redirect? If we want to solve for that scenario too, we are going to have to abruptly break all usage of `gulp` for AMP with no deprecation notice and no warning.

I'm inclined to go with the less invasive node script approach with the warning. WDYT?

Reply:
I'm inclined to go with the less invasive postinstall script approach with the warning. WDYT?

I think this is a good compromise. Ship it :)

Reply:
Added a `postinstall` script. Installation of the `gulp` fallback is less invasive now, and everything is much clearer. We also skip this during CI to keep things simple. Glad there was pushback on this.